drivers/ethernet/eth_mcux.c

```diff
@@ -157,22 +157,35 @@ struct eth_context {
 #endif
 	enet_handle_t enet_handle;
 #if defined(CONFIG_PTP_CLOCK_MCUX)
+	struct net_pkt *ts_tx_pkt;
 	const struct device *ptp_clock;
 	enet_ptp_config_t ptp_config;
 	float clk_ratio;
+	struct k_mutex ptp_mutex;
 #endif
 	struct k_sem tx_buf_sem;
 	phy_handle_t *phy_handle;
+	struct k_sem rx_thread_sem;
+	struct k_sem tx_thread_sem;
 	enum eth_mcux_phy_state phy_state;
 	bool enabled;
 	bool link_up;
 	uint32_t phy_addr;
+	uint32_t rx_irq_num;
+	uint32_t tx_irq_num;
 	phy_duplex_t phy_duplex;
 	phy_speed_t phy_speed;
 	uint8_t mac_addr[6];
 	void (*generate_mac)(uint8_t *);
 	struct k_work phy_work;
 	struct k_work_delayable delayed_phy_work;
+
+	K_KERNEL_STACK_MEMBER(rx_thread_stack, 1600);
```

Comment:
Please avoid magic constants, use a `#define` instead here for the size.

Reply:
Thank you for your suggestion. I have now used a define for this. Please, mark the conversation as resolved if you like my new proposal.